static/app/views/dashboards/dashboardLegendUtils.tsx

```diff
@@ -0,0 +1,204 @@
+        Object.keys(selected).includes(`Releases:${widget.id}`);
+
+      const thisWidgetWithoutReleasesWasSelected =
+        !Object.keys(selected).includes(`Releases:${widget.id}`) &&
+        Object.values(selected).filter(value => value === false).length === 1;
+
+      if (thisWidgetWithReleasesWasSelected || thisWidgetWithoutReleasesWasSelected) {
+        router.replace({
+          query: {
+            ...location.query,
+            unselectedSeries: newLegendQuery,
+          },
+        });
+      }
+    } else if (Array.isArray(location.query.unselectedSeries)) {
+      let isInQuery = false;
+      newLegendQuery = location.query.unselectedSeries.map(widgetLegend => {
+        if (widgetLegend.includes(widget.id!)) {
+          isInQuery = true;
+          // names of legend options are stored as seriesName:widgetId and are stored in
+          // query param as widgetId-seriesName-seriesName2-...
+          return this.encodeLegendQueryParam(widget, selected);
+        }
+        return widgetLegend;
+      });
+
+      !isInQuery
+        ? router.replace({
+            query: {
+              ...location.query,
+              unselectedSeries: [
+                ...location.query.unselectedSeries,
+                this.encodeLegendQueryParam(widget, selected),
+              ],
+            },
+          })
+        : router.replace({
+            query: {
+              ...location.query,
+              unselectedSeries: newLegendQuery,
+            },
+          });
+    } else {
+      router.replace({
+        query: {
+          ...location.query,
+          unselectedSeries: [
+            location.query.unselectedSeries,
+            this.encodeLegendQueryParam(widget, selected),
+          ],
+        },
+      });
+    }
+  }
+
+  // sets unselected legend options by the legend query param
+  getLegendUnselected(widget: Widget) {
+    const location = this.location;
+
+    return location.query.unselectedSeries
+      ? this.decodeLegendQueryParam(widget)
+      : this.widgetRequiresLegendUnselection(widget) &&
+          this.organization.features.includes('dashboards-releases-on-charts')
+        ? {[this.encodeSeriesNameForLegend('Releases', widget.id)]: false}
+        : {};
+  }
+
+  widgetRequiresLegendUnselection(widget: Widget) {
+    return (
+      widget.displayType === DisplayType.AREA || widget.displayType === DisplayType.LINE
+    );
+  }
+
+  formatLegendDefaultQuery(widgetId?: string) {
+    return this.organization.features.includes('dashboards-releases-on-charts')
+      ? `${widgetId}-Releases`
+      : `${widgetId}-`;
+  }
+
+  // going from selected to query param
+  encodeLegendQueryParam(widget: Widget, selected: Record<string, boolean>) {
+    return (
+      widget.id +
+      '-' +
+      Object.keys(selected)
+        .filter(key => !selected[key])
+        .map(series => this.decodeSeriesNameForLegend(series))
+        .join('-')
```

Comment:
A little spooky to delimit both the widget ID vs. series names _and_ separate series names with `-`. Maybe a comma instead, to separate the series?

Reply:
comma isn't a good delimiter because there are commas in some series names, especially those with conditions. Ex: `count_if(transaction.duration,equals,300)`. Open to other delimiter suggestions though

Reply:
Oh good point, but that's true of `"-"`, no? They can appear in equation series! I think this might be a good reminder to URLEncode (and decode) the values 🤔

Reply:
ah yes you're right I didn't catch that. Ok so `encodeURIComponent()` seems to encode commas. I'm thinking of encoding the commas of the series names and delimiting them with a comma. As for delimiting with the widget ID, I want to refrain from using `'-'` as it will cause problems with equation series. Any thoughts on another delimiter? I was thinking of `:` or `;`

Reply:
`:` seems natural!